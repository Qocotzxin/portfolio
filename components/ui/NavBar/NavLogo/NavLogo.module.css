.outterCircle {
    animation: dash-forward 1s linear forwards;
}

.innerCircle {
    animation: dash-backward 1.2s linear forwards;
}

.text {
    opacity: 1;
    font-size: 1.25rem;
    font-family: Oswald, serif;
    text-transform: uppercase;
}

@keyframes dash-forward {
    to {
        stroke-dashoffset: 0;
    }
}

@keyframes dash-backward {
    to {
        stroke-dashoffset: 280;
    }
}