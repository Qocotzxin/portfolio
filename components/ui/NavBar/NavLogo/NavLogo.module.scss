.NavLogo {
  &__outterCircle {
    animation: dash-forward 1s linear forwards;
  }

  &__innerCircle {
    animation: dash-backward 1.2s linear forwards;
  }

  &__text {
    opacity: 1;
    font-size: 1.25rem;
    font-family: Oswald, serif;
    text-transform: uppercase;
  }
}

@keyframes dash-forward {
  to {
    stroke-dashoffset: 0;
  }
}

@keyframes dash-backward {
  to {
    stroke-dashoffset: 280;
  }
}
