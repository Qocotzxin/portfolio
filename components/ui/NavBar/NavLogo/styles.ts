import styles from "./NavLogo.module.scss";

export const Classes = {
  base: styles.NavLogo,
  outterCircle: styles.NavLogo__outterCircle,
  innerCircle: styles.NavLogo__innerCircle,
  text: styles.NavLogo__text,
};
