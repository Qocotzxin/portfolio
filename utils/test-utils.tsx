import { ChakraProvider } from "@chakra-ui/react";
import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import theme from "@theme";
import { HygraphProvider } from "contexts/hygraphContext";
import { Components, HygraphData } from "@models/hygraph";

export const hygraphMockData: Partial<HygraphData> = {
  skipLinks: [
    {
      text: "Skip to content",
    },
  ],
  ariaLabels: [
    {
      id: "1",
      metadata: "dark",
      component: Components.ColorModeSwitcher,
      content: "Switch to light mode",
    },
    {
      id: "2",
      metadata: "light",
      component: Components.ColorModeSwitcher,
      content: "Switch to dark mode",
    },
  ],
  navLinks: [
    {
      url: "test",
      order: 1,
      text: "Test",
    },
  ],
  languages: [
    { code: "en", displayName: "english", isActive: true },
    { code: "es", displayName: "spanish", isActive: true },
    { code: "nl", displayName: "dutch", isActive: false },
  ],
  footerLinks: [
    {
      url: "https://github.com",
      isActive: true,
      icon: "github",
      ariaLabel: "Visit Github",
    },
    {
      url: "https://test.com",
      isActive: false,
      icon: "github",
      ariaLabel: "Visit test",
    },
  ],
};

interface AllTheProvidersProps {
  hygraphData?: Partial<HygraphData>;
}

export const AllTheProviders: FC<PropsWithChildren & AllTheProvidersProps> = ({
  children,
  hygraphData = hygraphMockData,
}) => {
  return (
    <ChakraProvider theme={theme}>
      <HygraphProvider hygraphData={hygraphData}>{children}</HygraphProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  wrapperProps?: AllTheProvidersProps
) =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllTheProviders {...wrapperProps}>{children}</AllTheProviders>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { customRender as render };
