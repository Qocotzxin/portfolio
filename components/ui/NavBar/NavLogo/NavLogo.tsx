import { FC } from "react";
import styles from "./NavLogo.module.css";

interface NavLogoProps {
  withAnimation?: boolean;
}

const NavLogo: FC<NavLogoProps> = ({ withAnimation }) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="48"
      height="48"
      viewBox="0 0 48 48"
      aria-labelledby="title"
      role="presentation"
    >
      <title id="title" lang="en">
        Logo
      </title>
      <linearGradient
        id="gradient"
        x1="0"
        x2="0"
        y1="0"
        y2="100%"
        gradientUnits="userSpaceOnUse"
      >
        <stop stopColor="#4FD1C5" offset="40%" />
        <stop stopColor="#4299E1" offset="70%" />
      </linearGradient>
      <circle
        cx="24"
        cy="24"
        r="20"
        data-testid="NavLogo-outter-circle"
        className={withAnimation ? styles.outterCircle : ""}
        fill="none"
        stroke="#4FD1C5"
        strokeWidth="1"
        strokeDasharray="128"
        strokeDashoffset="128"
      />
      <circle
        cx="24"
        cy="24"
        r="16"
        data-testid="NavLogo-inner-circle"
        className={withAnimation ? styles.innerCircle : ""}
        fill="none"
        stroke="#4299E1"
        strokeWidth="1"
        strokeDasharray="128"
        strokeDashoffset="128"
      />
      <text
        x="14"
        y="32"
        fill="url(#gradient)"
        data-testid="NavLogo-text"
        className={`${styles.text} ${withAnimation ? "fadeIn-regular" : ""}`}
      >
        CE
      </text>
    </svg>
  );
};

export default NavLogo;
