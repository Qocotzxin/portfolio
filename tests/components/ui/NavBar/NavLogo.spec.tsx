import NavLogo from "@ui/NavBar/NavLogo";
import { render, screen } from "@utils/test-utils";

describe("NavLogo", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<NavLogo withAnimation />);
    expect(container).toMatchSnapshot();
  });

  it("Should have animation related classes when withAnimation is true.", () => {
    render(<NavLogo withAnimation />);

    expect(screen.getByTestId("NavLogo-outter-circle")).toHaveClass(
      "outterCircle"
    );

    expect(screen.getByTestId("NavLogo-inner-circle")).toHaveClass(
      "innerCircle"
    );

    expect(screen.getByTestId("NavLogo-text")).toHaveClass("textAnimation");
  });

  it("Should NOT have animation related classes when withAnimation is false or undefined.", () => {
    render(<NavLogo />);

    expect(screen.queryByTestId("NavLogo-outter-circle")).not.toHaveClass(
      "outterCircle"
    );

    expect(screen.queryByTestId("NavLogo-inner-circle")).not.toHaveClass(
      "innerCircle"
    );

    expect(screen.queryByTestId("NavLogo-text")).not.toHaveClass(
      "textAnimation"
    );
  });
});
