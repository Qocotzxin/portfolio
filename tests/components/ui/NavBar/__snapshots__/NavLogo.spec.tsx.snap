// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLogo Should match snapshot. 1`] = `
<div>
  <svg
    aria-labelledby="title"
    height="48"
    role="presentation"
    viewBox="0 0 48 48"
    width="48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="title"
      lang="en"
    >
      Logo
    </title>
    <lineargradient
      gradientUnits="userSpaceOnUse"
      id="gradient"
      x1="0"
      x2="0"
      y1="0"
      y2="100%"
    >
      <stop
        offset="40%"
        stop-color="#4FD1C5"
      />
      <stop
        offset="70%"
        stop-color="#4299E1"
      />
    </lineargradient>
    <circle
      class="outterCircle"
      cx="24"
      cy="24"
      data-testid="NavLogo-outter-circle"
      fill="none"
      r="20"
      stroke="#4FD1C5"
      stroke-dasharray="128"
      stroke-dashoffset="128"
      stroke-width="1"
    />
    <circle
      class="innerCircle"
      cx="24"
      cy="24"
      data-testid="NavLogo-inner-circle"
      fill="none"
      r="16"
      stroke="#4299E1"
      stroke-dasharray="128"
      stroke-dashoffset="128"
      stroke-width="1"
    />
    <text
      class="text textAnimation"
      data-testid="NavLogo-text"
      fill="url(#gradient)"
      x="14"
      y="32"
    >
      CE
    </text>
  </svg>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
